sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -0,0 +1,968 @@
+      RecordReader<?, ?> reader = inputFormatObj.createRecordReader(inputSplits.get(0).getSplit(),
+          new TaskAttemptContextImpl(conf.getHadoopConfiguration(), new TaskAttemptID()));
+      if (reader == null) {
+        throw new IOException(
+            String.format(HadoopInputFormatIOContants.NULL_CREATE_RECORDREADER_ERROR_MSG,
+                inputFormatObj.getClass()));
+      }
+      reader.initialize(inputSplits.get(0).getSplit(),
+          new TaskAttemptContextImpl(conf.getHadoopConfiguration(), new TaskAttemptID()));
+      reader.nextKeyValue();
+    }
+
+   private void validateClass(Class<?> expectedClass, String property, String errorMessage){
+      Class<?> actualClass = conf.getHadoopConfiguration().getClass(property, Object.class);
+      if (actualClass != expectedClass) {
+        throw new IllegalArgumentException(
+            String.format(errorMessage,expectedClass.getName(),actualClass.getName()));
+      }
+    }
+
+    /**
+     * Returns InputFormat object.
+     */
+    @VisibleForTesting
+    InputFormat<?, ?> getInputFormat(){
+      return inputFormatObj;
+    }
+
+    @Override
+    public Coder<KV<K, V>> getDefaultOutputCoder() {
+      return KvCoder.of(keyCoder, valueCoder);
+    }
+
+    @Override
+    public BoundedReader<KV<K, V>> createReader(PipelineOptions options) throws IOException {
+      this.validate();
+      if (inputSplit == null) {
+          throw new IOException(HadoopInputFormatIOContants.CREATEREADER_UNSPLIT_SOURCE_ERROR_MSG);
+      } else {
+        return new HadoopInputFormatReader<Object>(this, keyTranslationFunction,
+            valueTranslationFunction, inputFormatObj, inputSplit.getSplit());
+      }
+    }
+
+    /** BoundedReader for HadoopInputFormatSource. */
+    class HadoopInputFormatReader<T extends Object> extends BoundedSource.BoundedReader<KV<K, V>> {
+
+      private final HadoopInputFormatBoundedSource<K, V> source;
+      private final InputSplit split;
+      @Nullable private final SimpleFunction<T, K> keyTranslationFunction;
+      @Nullable private final SimpleFunction<T, V> valueTranslationFunction;
+      private volatile boolean doneReading = false;
+      private long recordsReturned = 0L;
+      private InputFormat<?, ?> inputFormatObj;
+
+      public HadoopInputFormatReader(HadoopInputFormatBoundedSource<K, V> source,
+          @Nullable SimpleFunction keyTranslationFunction,
+          @Nullable SimpleFunction valueTranslationFunction,
+          InputFormat<?, ?> inputFormatObj,
+          InputSplit split) {
+        this.source = source;
+        this.keyTranslationFunction = keyTranslationFunction;
+        this.valueTranslationFunction = valueTranslationFunction;
+        this.inputFormatObj = inputFormatObj;
+        this.split = split;
+      }
+
+      @Override
+      public HadoopInputFormatBoundedSource<K, V> getCurrentSource() {
+        return source;
+      }
+
+      private RecordReader<T, T> currentReader;
```

Comment:
Putting all fields together.  these variables were added here as they were being used in the functions below, they are however not local variables, so we will declare them together at the same place